docs/sources/setup-grafana/configure-docker.md

```diff
@@ -57,48 +91,217 @@ The following settings are hard-coded when launching the Grafana Docker containe
+
+Example:
+
+The following command runs Grafana Enterprise on **port 3000** in detached mode and installs the custom plugin, which is specified as a URL parameter in the `GF_INSTALL_PLUGINS` environment variable.
+
+```bash
+docker run -d
+  -p 3000:3000
+  --name=grafana
+  -e "GF_INSTALL_PLUGINS=http://plugin-domain.com/my-custom-plugin.zip;custom-plugin,grafana-clock-panel"
+  grafana/grafana-enterprise
+```
+
+## Build and run a Docker image with pre-installed plugins
+
+To create multiple images with the same plugins, you can save time by building your own customized image that includes those plugins.
+
+In the Grafana GitHub repository, the `packaging/docker/custom/` folder includes a `Dockerfile` that you can use to build a custom Grafana image. The `Dockerfile` accepts `GRAFANA_VERSION`, `GF_INSTALL_PLUGINS`, and `GF_INSTALL_IMAGE_RENDERER_PLUGIN` as build arguments.
+
+The `GRAFANA_VERSION` build argument must be a valid `grafana/grafana` Docker image tag. By default, Grafana builds an Alpine-based image. To build an Ubuntu-based image, append `-ubuntu` to the `GRAFANA_VERSION` build argument.
+
+Example:
+
+The following example shows you how to build and run a custom Grafana Docker image based on the latest official Ubuntu-based Grafana Docker image:
+
+```bash
+# go to the custom directory
+cd packaging/docker/custom
+
+# run the docker build command to build the image
+docker build
+  --build-arg "GRAFANA_VERSION=latest-ubuntu"
+  -t grafana-custom .
+
+# run the custom grafana container using docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins
+
+To save time, you can customize a Grafana image by including plugins available on the [Grafana Plugin download page](/grafana/plugins). By doing so, you won't have to manually install the plugins each time, making the process more efficient.
+
+> **Note:** To specify the version of a plugin, you can use the `GF_INSTALL_PLUGINS` build argument and add the version number. The latest version is used if you don't specify a version number. For example, you can use `--build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel 1.0.1,grafana-simple-json-datasource 1.3.5"` to specify the versions of two plugins.
+
+Example:
+
+The following example shows how to build and run a custom Grafana image with pre-installed plugins.
+
+```bash
+# go to the custom directory
+cd packaging/docker/custom
+
+# running the build command
+# include the plugins you want e.g. clock planel etc
+docker build
+  --build-arg "GRAFANA_VERSION=latest"
+  --build-arg "GF_INSTALL_PLUGINS=grafana-clock-panel,grafana-simple-json-datasource"
+  -t grafana-custom .
+
+# running the custom Grafana container using the docker run command
+docker run -d -p 3000:3000 --name=grafana grafana-custom
+```
+
+## Build a Grafana Docker image with pre-installed plugins from other sources
```

Comment:
This section doesn't actually show the user how to preinstall a plugin from another source, which requires prefixing the plugin slug with the url of a custom .zip file